### PR TITLE
HDDS-8193. Ozone file systems to support Hadoop's PathCapabilities interface

### PR DESCRIPTION
**Pull request by jojochuang**
## What changes were proposed in this pull request?

Implemented PathCapabilities interface APIs in ofs/o3fs.

Three existing capabilities are added: FS_ACLS, FS_CHECKSUMS and FS_SNAPSHOTS. This is implemented following the utility class DfsPathCapabilities in Hadoop HDFS.

The lease recovery and safemode capabilities are not yet included. Will add in a future jira.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8193

## How was this patch tested?
Unit tests

### CODE REVIEW

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzonePathCapabilities.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   * @param capability capability
+   * @return either a value to return or, if empty, a cue for the FS to
+   * pass up to its superclass.
+   */
+  public static Optional<Boolean> hasPathCapability(final Path path,
+      final String capability) {
+    switch (validatePathCapabilityArgs(path, capability)) {
+    case CommonPathCapabilities.FS_ACLS:
+    case CommonPathCapabilities.FS_CHECKSUMS:
+    case CommonPathCapabilities.FS_SNAPSHOTS:
```
Comment (szetszwo) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzonePathCapabilities.java:48]: Ozone seems not yet support snapshots?

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzonePathCapabilities.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      return Optional.of(true);
+    default:
+      return Optional.empty();
```
Comment (szetszwo) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzonePathCapabilities.java:51]: Let's simply return boolean.  BTW, just filed https://issues.apache.org/jira/browse/HDFS-16963 to fix HDFS.

**Review by szetszwo — COMMENTED** [review]
@jojochuang , thanks a lot for working on this!  Please see the comments inlined.

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzonePathCapabilities.java (hunk 1 above)
Comment (jojochuang) [reply hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzonePathCapabilities.java:48]: createSnapshot() is implemented in BasicRootedOzoneFileSystem.
But you're right. The rest of snapshot APIs are not supported yet.

Once snapshot is supported, the path capability can be added. Also will need to update contract tests.

**Review by smengcl — APPROVED** [review]
lgtm. Thanks @jojochuang !

**Pull request merged**